### ISSUE TITLE
Add support to DB 11.3 ML LTS in databricks build script

### ISSUE THREAD

**Issue by amahussein** [NVIDIA/spark-rapids#6979]
While Working on adding support to DB ML LTS, I found the following issues with jenkins/build.sh

- The jar files for DB 11.3 has different pattern/prefix. The current implementation is hardcoding the pattern.
- Hardcoded software versions. 
- Unused variables in the script. Example `JARDIR=/databricks/jars` and `DB_JAR_LOC=/databricks/jars/` and `COREPOM=spark-core_${SCALA_VERSION}-${BASE_SPARK_VERSION}.pom` and `COREPOMPATH=$M2DIR/org/apache/spark/spark-core_${SCALA_VERSION}/${BASE_SPARK_VERSION}`
- Some JARs are redundant. For example the protobuf jar is repeated twice `$PROTOBUF_JAR` and `$PROTOBUFJAVA`.
- Some JARS are defined twice.

**In summary, we need to:**

- fix the code so that the prefix of the jar names are not hardcoded.
- remove hardcoded versions from the code.
- add spark-3.3.0 jars to the accepted values.
- It will be nice if the the code has separate sections to make it easier to extend. For example, define jars in one function rather than having definitions all over the place.
- `set -ex` makes following the output difficult. Instead, it will be better to have the `-x` enabled for the main commands.

**Comment by amahussein** [comment]
related https://github.com/NVIDIA/spark-rapids/pull/7097
